### PR TITLE
Support javascript strict mode for tests (use strict)

### PR DESCRIPTION
**Pull request by Joris-van-der-Wel**
See #95 

Anything I forgot? (I am not sure how to generate the readme)

### CODE REVIEW

**Comment by jdalton** [comment]
Would you make the spelling corrections a separate PR?

File: benchmark.js [hunk 1]
```diff
@@ -2206,6 +2207,14 @@
         'defer': false,
 
         /**
+         * A flag to indicate that the test should run in javascript strict mode.
+         *
+         * @memberOf Benchmark.options
+         * @type boolean
+         */
+        'strictMode': false,
```
Comment (jdalton) [inline benchmark.js]: Instead of `strictMode` can you make it `strict`.

**Comment by Joris-van-der-Wel** [comment]
updated

**Comment by kevinoid** [comment]
I just came across this PR while looking for strict mode support.  Was there something preventing it from being merged?  Why was it closed?  I suspect I'm missing some context.

**Comment by Joris-van-der-Wel** [comment]
Oh sorry, I closed it because I was cleaning up my `/pulls`. In the project it was for I work around the whole compilation stuff anyway: https://github.com/tmpvar/jsdom/blob/master/benchmark/jsdom-benchmark.js

**Comment by kevinoid** [comment]
@Joris-van-der-Wel Thanks for the explanation and the link to the code!

Moving the test body into a function seems like it might work against the stringify/recompile steps done by benchmark.js to avoid reaching optimization thresholds.  I suppose that only really matters for microbenchmarks, since calling any other non-native functions would have the same behavior, but I don't have enough expertise to know for sure.

Thanks again for the update and code!

**Comment by Uzlopak** [comment]
I will adapt this for my fork. Thanks

**Pull request closed without merging**